mathics/builtin/inout.py

```diff
@@ -206,10 +206,14 @@ def parenthesize(precedence, element, element_boxes, when_equal):
 
     while element.has_form("HoldForm", 1):
         element = element.elements[0]
+
     if element.has_form(("Infix", "Prefix", "Postfix"), 3, None):
         element_prec = element.elements[2].get_int_value()
     elif element.has_form("PrecedenceForm", 2):
         element_prec = element.elements[1].get_int_value()
+    # For negative values, ensure that the element_precedence is at least the precedence. (Fixes #332)
+    elif isinstance(element, Integer) and element.value < 0:
+        element_prec = precedence
```

Comment:
I'm not sure about this, this really seems a bug in parser that is passed forward and fixed somewhere later. @rocky @mmatera ?

Reply:
@TiagoCavalcante, is not a problem in the parser, but in the way expressions involving Power are formatted:
`Power[a,b]` is formatted as `a^b`, while `Power[a c,b]` is formatted as `(a c)^b`. However, negative numbers must be handled different: `Power[-1, b]` should be formatted as `(-1)^b`, but in master, it is formatted as a with any other positive number  `-1^b`. This is what this PR is fixing.

Reply:
Now I understand, thanks for the explanation!

Reply:
But is this simply limited to Integers? Wouldn't there be the same problem with any sort of _numeric_ literal, like -5.3 ? 

The other thing I wonder about is under exactly which conditions it is needed. It is needed for Infix of which StandardForm and other Forms follow. But plain Infix is above in line 209 or 210 and this doesn't hit that. 

(The get_int_value()'s could be simplified.) 

I suspect we are also going to have to make another pass over this code because we "canonicalize" or order Polynomials the opposite order, but fixing that is going to cause lots of changes in the expected output behavior so that will have to be deferred.

Reply:
> But is this simply limited to Integers? Wouldn't there be the same problem with any sort of _numeric_ literal, like -5.3 ?
> 

Thanks for noting this!

> The other thing I wonder about is under exactly which conditions it is needed. It is needed for Infix of which StandardForm and other Forms follow. But plain Infix is above in line 209 or 210 and this doesn't hit that.
> 

I have to check it...

> (The get_int_value()'s could be simplified.)
> 

I changed this in this part of the code, but it appears in many places of the module. I am going to hold off changing the other cases until we face the formatting refactor.

> I suspect we are also going to have to make another pass over this code because we "canonicalize" or order Polynomials the opposite order, but fixing that is going to cause lots of changes in the expected output behavior so that will have to be deferred.

Yep. This is another scale.